SQL/SQL.sublime-syntax

```diff
@@ -259,12 +306,28 @@ contexts:
       pop: true
     - match: (?=\S)
       pop: true
-  like_string:
-    - match: \'
-      scope: punctuation.definition.string.begin.sql
-      set: inside_like_single_quoted_string
+  like_escape_slash:
+    - include: comments
+    - match: (\')(\\)(\')
+      captures:
+        0: meta.string.escape.sql string.quoted.single.sql
+        1: punctuation.definition.string.begin.sql
+        2: constant.character.escape.sql
+        3: punctuation.definition.string.end.sql
+      pop: true
     - match: (?=\S)
+      fail: like_string_branch
+  like_escape_caret:
+    - include: comments
+    - match: (\')(\^)(\')
+      captures:
+        0: meta.string.escape.sql string.quoted.single.sql
+        1: punctuation.definition.string.begin.sql
+        2: constant.character.escape.sql
+        3: punctuation.definition.string.end.sql
       pop: true
+    - match: (?=\S)
+      fail: like_string_branch
   inside_like_single_quoted_string:
```

Comment:
I found it useful to keep those body contexts grouped together with those which push them on stack, especially if they are used once or a few times only.